### PR TITLE
Add authorization by username and password

### PR DESCRIPTION
**Pull request by antondevs**
Please add authorization by username and password

### CODE REVIEW

**Comment by phimage** [comment]
Thanks
@antondevs did you have some exemple of service/provider that use this authorization mode?
Better if I can create a free account

Then the service could be added to the demo

File: Sources/OAuth2Swift.swift [hunk 1]
```diff
@@ -287,6 +287,25 @@ open class OAuth2Swift: OAuthSwift {
...
+    open func authorize(username: String, password:String, success: @escaping TokenRenewedHandler, failure: @escaping OAuthSwiftHTTPRequest.FailureHandler) -> OAuthSwiftRequestHandle? {
+        var parameters = OAuthSwift.Parameters()
+        parameters["client_id"] = self.consumerKey
+        parameters["client_secret"] = self.consumerSecret
+        parameters["username"] = username
+        parameters["password"] = password
+        parameters["grant_type"] = "password"
+    
+        print(parameters)
+
```
Comment (phimage) [inline Sources/OAuth2Swift.swift]: no print in code

File: Sources/OAuth2Swift.swift [hunk 2]
```diff
@@ -287,6 +287,25 @@ open class OAuth2Swift: OAuthSwift {
             }
         }
     }
+	
+	@discardableResult
+    open func authorize(username: String, password:String, success: @escaping TokenRenewedHandler, failure: @escaping OAuthSwiftHTTPRequest.FailureHandler) -> OAuthSwiftRequestHandle? {
+        var parameters = OAuthSwift.Parameters()
```
Comment (phimage) [inline Sources/OAuth2Swift.swift]: why `TokenRenewedHandler` ? copied from renew function
maybe SuccessHandler

**Comment by phimage** [comment]
please put in function comment the link about the rfc

then could you add also optional `scope: String?` as the rfc describe 
and also `headers: OAuthSwift.Headers? = nil`

**Comment by antondevs** [comment]
Updated. I added headers and comment. Scope param is not exist in requestOAuthAccessToken. Please apply

**Comment by antondevs** [comment]
Please review. I am not using scope, also added scope param to func

**Pull request merged**